src/backend/PhotoAPI/DataAccess/Implementations/PhotoBlobStorage.cs

```diff
@@ -2,68 +2,110 @@
-        {
-            CloudStorageAccount cloudStorageAccount = CloudStorageAccount.Parse(settings.ConnectionString);
-            CloudBlobClient cloudBlobClient = cloudStorageAccount.CreateCloudBlobClient();
-
-            _cloudBlobContainerPhotos = CreateContainer(settings, cloudBlobClient);
-        }
-
-        private CloudBlobContainer CreateContainer(BlobStorageSettings settings, CloudBlobClient cloudBlobClient)
-        {
-            CloudBlobContainer cloudBlobContainerPhotos = cloudBlobClient.GetContainerReference(settings.ImageContainerName);
-            cloudBlobContainerPhotos.CreateIfNotExists();
-
-            BlobContainerPermissions permissions = new BlobContainerPermissions
-            {
-                PublicAccess = BlobContainerPublicAccessType.Blob
-            };
-
-            cloudBlobContainerPhotos.SetPermissionsAsync(permissions);
-            return cloudBlobContainerPhotos;
-        }
-
-        // METHODS
-        public async Task<string> UploadFileAsync(byte[] file, string contentType)
-        {
-            string extension = GetExtension(contentType);
-            string fileName = System.Guid.NewGuid().ToString() + extension;
-
-            CloudBlockBlob block = _cloudBlobContainerPhotos.GetBlockBlobReference(fileName);
-            block.Properties.ContentType = contentType;
-
-            await block.UploadFromByteArrayAsync(file, 0, file.Length);
-
-            return GetFullBlobName(block);
-        }
-
-        private string GetExtension(string contentType)
-        {
-            return MimeTypes.MimeTypeMap.GetExtension(contentType);
-        }
-
-        private string GetFullBlobName(CloudBlockBlob block)
-        {
-            return block.Uri.AbsoluteUri;
-        }
-
-        public Task<bool> DeleteFileIfExistsAsync(string blobName)
-        {
-            CloudBlockBlob blob = _cloudBlobContainerPhotos.GetBlockBlobReference(blobName);
-            return blob.DeleteIfExistsAsync();
-        }
-        
-    }
+	public class PhotoBlobStorage : Interfaces.IPhotoBlobStorage
+	{
+		// FIELDS
+		private readonly CloudBlobContainer _cloudBlobContainerPhotos;
+
+		// CONSTRUCTORS
+		public PhotoBlobStorage(BlobStorageSettings settings)
+		{
+			CloudStorageAccount cloudStorageAccount = CloudStorageAccount.Parse(settings.ConnectionString);
+			CloudBlobClient cloudBlobClient = cloudStorageAccount.CreateCloudBlobClient();
+
+			ConfigureCors(cloudBlobClient);
+
+			_cloudBlobContainerPhotos = CreateContainer(settings, cloudBlobClient);
+		}
+
+		private void ConfigureCors(CloudBlobClient cloudBlobClient)
+		{
+			ServiceProperties serviceProperties = cloudBlobClient.GetServiceProperties();
+
+			CorsProperties corsProperties = new CorsProperties();
+			corsProperties.CorsRules.Add(new CorsRule()
+			{
+				AllowedMethods = CorsHttpMethods.Get,
+				AllowedHeaders = new string[] { "*" },
+				AllowedOrigins = new string[] { "*" },
+				ExposedHeaders = new string[] { "*" },
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/iamprovidence/Lama/pullRequest?prid=5141404)